compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/v2/Scrollbar.desktop.kt

```diff
@@ -463,19 +454,39 @@ internal class SliderAdapter(
         unscrolledDragDistance = 0.0
     }
 
-    /** Called on every movement while dragging the thumb */
-    fun onDragDelta(offset: Offset) {
-        val dragDelta = if (isVertical) offset.y else offset.x
-        val maxScrollPosition = adapter.maxScrollOffset * scrollScale
-        val currentPosition = position
-        val targetPosition =
-            (currentPosition + dragDelta + unscrolledDragDistance).coerceIn(0.0, maxScrollPosition)
-        val sliderDelta = targetPosition - currentPosition
+    private suspend fun setRawPosition(value: Double) {
+        adapter.scrollTo(value / scrollScale)
+    }
 
-        // Have to add to position for smooth content scroll if the items are of different size
-        position += sliderDelta
+    private val setRawPositionMutex = Mutex()
 
-        unscrolledDragDistance += dragDelta - sliderDelta
+    /** Called on every movement while dragging the thumb */
+    fun onDragDelta(offset: Offset) {
+        coroutineScope.launch {
```

Comment:
How about `launch(start = CoroutineStart.UNDISPATCHED)` here to preserve the current behavior for scrollers that don't suspend?

Reply:
Right. done